libraries/AP_ADSB/AP_ADSB_Sagetech.cpp

```diff
@@ -0,0 +1,746 @@
+    for (uint8_t i=0; i<sizeof(message_format_header); i++) {
+        gcs().send_text(MAV_SEVERITY_DEBUG, "%shead[%2u] = 0x%02x", frontend.GcsHeader, i, message_format_header[i]);
+    }
+#endif
+#if SAGETECH_DEBUG_TX_ID_ONLY
+    gcs().send_text(MAV_SEVERITY_DEBUG, "%sTx type=%d %s", frontend.GcsHeader, static_cast<uint8_t>(msg.type), type_to_str(msg.type));
+#endif
+#if SAGETECH_DEBUG_TX_ID_PAYLOAD || SAGETECH_DEBUG_TX_ALL_RAW
+    #if SAGETECH_DEBUG_TX_ID_PAYLOAD
+    gcs().send_text(MAV_SEVERITY_DEBUG, "%sTx type=%u, payload.len=%u", frontend.GcsHeader, (unsigned)msg.type, (unsigned)msg.payload_length);
+    #endif
+    for (uint8_t i=0; i<msg.payload_length; i++) {
+        const uint8_t data = msg.payload[i];
+        if (isalnum(data) || data == '.' || data == '-' || data == ' ') {
+            gcs().send_text(MAV_SEVERITY_DEBUG, "%sdata[%2u] = 0x%02x  %c", frontend.GcsHeader, i, data, (char)data);
+        } else {
+            gcs().send_text(MAV_SEVERITY_DEBUG, "%sdata[%2u] = 0x%02x", frontend.GcsHeader, i, data);
+        }
+    }
+#endif
+
+#if SAGETECH_DEBUG_TX_ALL_RAW
+    for (uint8_t i=0; i<sizeof(message_format_tail); i++) {
+        gcs().send_text(MAV_SEVERITY_DEBUG, "%stail[%2u] = 0x%02x", frontend.GcsHeader, (unsigned)(i + sizeof(message_format_header) + msg.payload_length), message_format_tail[i]);
+    }
+#endif
+}
+
+
+
+
+
+void AP_ADSB_Sagetech::send_Installation()
+{
+    uint8_t adsb_in_vehicles_per_second = 0;
+    uint8_t baud_value = 0;
+
+    switch (baudrate) {
+    case 2400:
+        baud_value = 1;
+        adsb_in_vehicles_per_second = 0;
+        break;
+    case 4800:
+        baud_value = 2;
+        adsb_in_vehicles_per_second = 0;
+        break;
+    case 9600:
+        baud_value = 3;
+        adsb_in_vehicles_per_second = 5;
+        break;
+    case 19200:
+        baud_value = 4;
+        adsb_in_vehicles_per_second = 10;
+        break;
+    case 38:
+    case 38400:
+        baud_value = 5;
+        adsb_in_vehicles_per_second = 20;
+        break;
+    default:
+    case 57:
+    case 57600:
+        baud_value = 0;
+        adsb_in_vehicles_per_second = 30;
+        break;
+    case 115:
+    case 115000:
+    case 115200:
+        baud_value = 6;
+        adsb_in_vehicles_per_second = 60;
+        break;
+    case 230:
+    case 230000:
+    case 230400:
+        baud_value = 7;
+        adsb_in_vehicles_per_second = 100;
+        break;
+    case 460:
+    case 460000:
+    case 460800:
+        baud_value = 8;
+        adsb_in_vehicles_per_second = 100;
+        break;
+    }
+
+    Packet_XP pkt {};
+
+    pkt.type = MsgTypes_XP::Installation_Set;
+    pkt.payload_length = 28; // 28== 0x1C
+
+    // Mode C = 3, Mode S = 0
+    pkt.id = (transponder_type == Transponder_Type::Mode_C) ? 3 : 0;
+
+
+
+
+//    // convert a decimal 123456 to 0x123456
+    // TODO: do a proper conversion. The param contains "131313" but what gets transmitted over the air is 0x200F1.

```

Comment:
This needs to be fixed before it goes in, right?  Otherwise we can't fix it later.